compiler/src/iree/compiler/Dialect/Flow/Transforms/OutlineDispatchRegions.cpp

```diff
@@ -65,12 +65,61 @@ static std::string getLinalgOpLoopRanges(linalg::LinalgOp op) {
   return outputString;
 }
 
+static std::string operandTypeToString(OpOperand *opOperand) {
+  auto operandType = opOperand->get().getType();
+  std::string outputString;
+  llvm::raw_string_ostream sstream(outputString);
+  if (auto shapedType = dyn_cast<ShapedType>(operandType)) {
+    shapedType.getElementType().print(sstream);
+  } else {
+    operandType.print(sstream);
+  }
+  return outputString;
+}
+
+// Returns a string like "f32xi32xf16" representing a linalg op's types for each
+// operands. Will collapse to single type if all match.
+static std::string getLinalgDataTypes(linalg::LinalgOp op) {
+  std::string firstToken = "";
+  bool allTokensSame = true;
+  SmallVector<std::string, 4> datatypeTokens;
+
+  for (OpOperand *opOperand : op.getDpsInputOperands()) {
```

Comment:
@ScottTodd would be helpful if there was a concise way to concat two lists when iterating. Couldn't find anything in LLVM. Perhaps you may know of a construct?

Reply:
Input and init operands can have different lengths, right? You could concat those lists then have only a single for loop. Or, rather than concat the lists themselves, you could join the iterators with [`llvm::concat_iterator`](https://github.com/llvm/llvm-project/blob/e76d236c1cebfb7d31f722d242bc927b79dcdb49/llvm/include/llvm/ADT/STLExtras.h#L1102-L1115) (found via "c++ concatenate two iterators" -> https://stackoverflow.com/questions/6747987/writing-an-iterator-that-makes-several-containers-look-as-one and "llvm join iterators" -> https://llvm.org/doxygen/classllvm_1_1concat__iterator.html)